lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -227,13 +468,45 @@ bool BootloaderCom::process()
             m_waitingForResponse = false;
             if(m_cmdProvider->next(m_currentCommand, m_currentResponse))
             {
-                if(true == m_flashManager.sendCommand(
-                    m_currentCommand->command,
-                    m_currentCommand->commandsize))
+                /*Check if the command provider's command type is for firmware programming.*/
+                if (FW_PROG == m_cmdProvider->getCmdType()) 

```

Comment:
This if doesn not belong here. This function is just handlgng cmd data exchanging. It must not know what cmds or handle them conditionally.

Reply:
It will be take care by the next commit.